### PR TITLE
feat: add metrics_port flag

### PR DESCRIPTION
**Pull request by rrylee**
# Description

add `--metrics-port` on run command

## Issue reference

Please reference the issue this PR will close: https://github.com/dapr/cli/issues/554

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Review by mukundansundar — APPROVED** [review]
lgtm

**Comment by mukundansundar** [comment]
@rrylee Can you add a negative test for the metrics port not being set and that a proper port is selected?

File: pkg/standalone/run.go [hunk 1]
```diff
@@ -73,17 +74,28 @@ func getDaprCommand(appID string, daprHTTPPort int, daprGRPCPort int, appPort in
...
 	daprCMD := binaryFilePath(defaultDaprBinPath(), "daprd")
-	metricsPort, err := freeport.GetFreePort()
-	if err != nil {
-		return nil, -1, -1, -1, err
-	}
 
-	args := []string{"--app-id", appID, "--dapr-http-port", fmt.Sprintf("%v", daprHTTPPort), "--dapr-grpc-port", fmt.Sprintf("%v", daprGRPCPort), "--log-level", logLevel, "--app-max-concurrency", fmt.Sprintf("%v", maxConcurrency), "--app-protocol", protocol, "--metrics-port", fmt.Sprintf("%v", metricsPort), "--components-path", componentsPath}
+	args := []string{
+		"--app-id", appID,
+		"--dapr-http-port", fmt.Sprintf("%v", daprHTTPPort),
```
Comment (tcnghia) [inline pkg/standalone/run.go:80]:
```suggestion
		"--dapr-http-port", strconv.Itoa(daprHTTPPort),
```
I think `fmt.Sprintf()` is both longer, less efficient, and more error-prone (ignore types). It's not the purpose of this PR but probably nice to include the clean up here. Same with other occurences.

File: pkg/standalone/run.go [hunk 2]
```diff
@@ -49,7 +50,7 @@ type RunOutput struct {
 	AppCMD       *exec.Cmd
 }
 
-func getDaprCommand(appID string, daprHTTPPort int, daprGRPCPort int, appPort int, configFile, protocol string, enableProfiling bool, profilePort int, logLevel string, maxConcurrency int, placementHost string, componentsPath string, appSSL bool) (*exec.Cmd, int, int, int, error) {
+func getDaprCommand(appID string, daprHTTPPort int, daprGRPCPort int, appPort int, configFile, protocol string, enableProfiling bool, profilePort int, logLevel string, maxConcurrency int, placementHost string, componentsPath string, appSSL bool, metricsPort int) (*exec.Cmd, int, int, int, error) {
```
Comment (tcnghia) [inline pkg/standalone/run.go:53]: Again, not the purpose of this PR but the code will be more readable if we have the output parameters named or documented in a doc comment.

**Comment by rrylee** [comment]
@tcnghia 
i will submit a pr for dapr/docs

**Review by tcnghia — DISMISSED** [review]

File: pkg/standalone/run.go [hunk 3]
```diff
@@ -185,7 +199,7 @@ func Run(config *RunConfig) (*RunOutput, error) {
 		return nil, err
 	}
 
-	daprCMD, daprHTTPPort, daprGRPCPort, metricsPort, err := getDaprCommand(appID, config.HTTPPort, config.GRPCPort, config.AppPort, config.ConfigFile, config.Protocol, config.EnableProfiling, config.ProfilePort, config.LogLevel, config.MaxConcurrency, config.PlacementHost, config.ComponentsPath, config.AppSSL)
+	daprCMD, daprHTTPPort, daprGRPCPort, metricsPort, err := getDaprCommand(appID, config.HTTPPort, config.GRPCPort, config.AppPort, config.ConfigFile, config.Protocol, config.EnableProfiling, config.ProfilePort, config.LogLevel, config.MaxConcurrency, config.PlacementHost, config.ComponentsPath, config.AppSSL, config.MetricsPort)
```
Comment (mukundansundar) [inline pkg/standalone/run.go:202]: @tcnghia  @yaron2 is there any reason not to pass the config struct to the getDaprCommand function as it it and extract all the fields in that method?

Comment (mukundansundar) [reply pkg/standalone/run.go:202]: @yaron2 any response on this question?

Comment (yaron2) [reply pkg/standalone/run.go:202]: There isn't, but it should happen in a separate PR.

**Review by yaron2 — APPROVED** [review]

**Comment by yaron2** [comment]
Thanks @rrylee!

**Pull request merged**